docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,715 @@
+The secrets discussed in this document are associated with accessing the data plane of various services.
+
+Think: Accessing a `StorageAccount` via Shared Key, accessing a `MySQLServer` by admin `Username` and `Password`, or accessing a VM
+by `SSHKey`.
+
+_Sometimes_ these secrets may be used by other CRDs managed by ASO, as would be the case for a `MySQLUser` CRD, but often the consumers of these
+secrets are the users applications.
+
+## Goals
+
+### ASO should not be generating or managing secrets on the users behalf
+
+ASO v1 generated secrets on the users behalf in some cases. The users had access to the generated secret as ASO wrote it into a Kubernetes (or KeyVault) 
+secret.
+
+There are problems with this approach:
+1. It makes performing secret rollover more difficult. If the user had specified the secret, rollover is as easy as modifying the specified secret, which 
+   triggers a reconcile, which triggers a PUT to the Azure Resource updating the secret. If the operator generated the secret then the user must somehow 
+   issue an instruction to us to perform an action to roll the secret, which doesn't fit as well into the goal-seeking paradigm Kubernetes prefers.
+2. It takes control away from the user. What if the secret the operator generates doesn't comply with a particular organizations complexity requirements?
+3. It requires that we do a _very_ good job generating cryptographically secure passwords/secrets. This could easily become a can of worms.
+4. It hinders adoption of existing resources. If the operator expects to always generate the secret for a SQL DB, but the user wants to import a SQL DB 
+   they've already created through some other mechanism, then they are at an impasse as there is no (easy) way for them to provide ASO with the secret.
+5. It doesn't work well with GitOps. A number of customers have expressed the desire to move resources between namespaces. In theory this is easy - 
+   just create the exact same resources in a different namespace (pointing at the same Azure resources), mark the old resources as `skip-deletion` (so that
+   the backing Azure resources are not deleted), and delete the original namespace. If the definition of the secret isn't part of the users GitOps flow this 
+   becomes more difficult as the first namespace has  the secret (which was created by ASO) and that secret must be cloned manually to the second namespace,
+   otherwise the creation of the resource in the second namespace will attempt to generate a new secret.
+
+We could investigate allowing both automatically generated secrets (the default) and user specified secrets (opt-in). That avenue results in the operator 
+shouldering all of the complexity burden of both though. 
+
+Given ASO's approach as a low level toolkit providing the ability to create Azure resources, at least 
+at this time we should avoid the added complexity of generating user secrets.
+
+### Prefer Managed Identity
```

Comment:
Why do want to give users an option of not using managed identity? Are we just limited by resources supporting managed identity or is there customer value in being able to choose?

Reply:
There are "good reasons" to not use managed identity. For example we use service principal for our ASO test environment, because managed identity requires you to run in Azure, which is annoying when you could otherwise kick the tires/test outside of Azure.

There are also some resources that don't support managed identity, or more often don't support it for all scenarios.

but I think it's _mostly_ that the service itself doesn't force you to use managed identity. If we for example forced all users of Azure storage to use managed identity, it would preclude users who just want to lift+shift their stuff into ASO and have it adopt their existing resources.

With that said, this preference for managed identity here means we shouldn't default to pulling secrets and we should probably (as best we can) document with managed identity first.